app/components/ui/SiteSearchInput.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { FormEvent } from "react";
+import { useHistory } from "react-router-dom";
+import cn from "classnames";
+import qs from "qs";
+import { useSiteSearch } from "../../hooks/SiteSearch";
+import styles from "./SiteSearchInput.module.scss";
+
+/**
+ * Sitewide listing search component
+ *
+ * - Updates the url querystring on every search
+ * - Allows empty searches by removing `query=` query param from querystring
+ */
+export const SiteSearchInput = () => {
+  const { query, setQuery } = useSiteSearch();
+  const history = useHistory();
+
+  const removeQueryStringStart = (querystring: string) => querystring.slice(1);
```

Comment:
@adriencyberspace I wasn't perceptive enough to understand what you were doing so I thought I'd wrap up (even this simple little operation) in a function behind an informative name.